File: C++/Medium/Accepted/number-of-matching-subsequences_7.cpp
class Solution {
public:
    int numMatchingSubseq(string s, vector<string>& words) {
        
        unordered_map<char, vector<string>> m;
        for (auto& w: words) {
            m[w[0]].push_back(w);
        }
        int res = 0;
        for (auto c: s) {
            vector<string> tmp = m[c];
            m[c].clear();
            for (auto& w: tmp) {
                if (w.size() == 1) {
                    res++;
                } else {
                    m[w[1]].push_back(w.substr(1));
                }
            }
        }
        return res;
    }
};

File: Python3/Medium/Accepted/number-of-matching-subsequences_7.py
class Solution:
    def numMatchingSubseq(self, s: str, words: List[str]) -> int:
        
        def isSubseq(s, t):
            i = 0
            for c in s:
                i = t.find(c, i) + 1
                if i == 0:
                    return False
            return True
        return sum(isSubseq(w, s) for w in words)